src/components/performances/PerformanceForm.jsx

```diff
@@ -19,20 +20,28 @@ class PerformanceForm extends React.Component {
     const { currentValues } = props;
 
     this.state = {
+      disabled: true,
       academicLevel: currentValues.academicLevel || '',
       choreographers: currentValues.choreographers || '',
       competitionLevel: currentValues.competitionLevel || '',
       danceEntry: currentValues.danceEntry,
       danceStyle: currentValues.danceStyle || '',
       danceTitle: currentValues.danceTitle || '',
-      disabled: false,
       options: {},
       performers: currentValues.performers || '',
       school: currentValues.school || '',
       size: currentValues.size || 0
     };
   }
 
+  // disable save button if not all input fields are filled
+  static getDerivedStateFromProps(props, state) {
+    const values = pick(state, 
+      ['academicLevel', 'choreographers', 'competitionLevel', 'danceEntry', 
```

Comment:
For these inline arrays, let's extract this into another const.

Reply:
Extracted into constant `fields`